### PR TITLE
Add snapshot checkpoint for E2E test

### PR DESCRIPTION
**Pull request by danfengliu**
Signed-off-by: danfengl <danfengl@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

1. Add checkpoint in snapshot E2E test to verify snapshot CR should be created and snapshot should be created in cloud side after backup completion;
2. Fix snapshot name issue that CSI snapshot name in cloud side is not the same with other non-CSI cloud snapshots.

# Does your change fix a particular issue?
#4804

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: test/e2e/util/velero/velero_utils.go [hunk 1]
```diff
@@ -797,3 +807,29 @@ func GetResticRepositories(ctx context.Context, veleroNamespace, targetNamespace
...
+
+func GetSnapshotCheckPoint(client TestClient, VeleroCfg VerleroConfig, expectCount int, namespaceBackedUp, backupName string, kibishiiPodNameList []string) (SnapshotCheckPoint, error) {
+	var snapshotCheckPoint SnapshotCheckPoint
+	snapshotCheckPoint.ExpectCount = expectCount
+	snapshotCheckPoint.NamespaceBackedUp = namespaceBackedUp
+	snapshotCheckPoint.PodName = kibishiiPodNameList
+	if strings.EqualFold(VeleroCfg.Features, "EnableCSI") {
+		snapshotCheckPoint.EnableCSI = true
+		if err := util.CheckVolumeSnapshotCR(client, kibishiiPodNameList, namespaceBackedUp, backupName); err != nil {
+			return snapshotCheckPoint, errors.Wrapf(err, "Fail to get Azure CSI snapshot content")
```
Comment (blackpiglet) [inline test/e2e/util/velero/velero_utils.go:819]: Looks like this is a common function for all cloud provider. May need to remove Azure from log information.

File: test/e2e/util/csi/common.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	return client, snapshotterClient, nil
+}
+
+func GetVolumeSnapshotContentNameByPod(client TestClient, podName, namespace, backupName string) (string, error) {
+	pvcList, err := GetPvcByPodName(context.Background(), namespace, podName)
+	if err != nil {
+		return "", err
+	}
+	if len(pvcList) != 1 {
```
Comment (blackpiglet) [inline test/e2e/util/csi/common.go:61]: It's possible one Pod has more than one PV, but please confirm whether this condition is acceptable for test environment.

File: test/e2e/util/velero/velero_utils.go [hunk 3]
```diff
@@ -797,3 +807,29 @@ func GetResticRepositories(ctx context.Context, veleroNamespace, targetNamespace
...
+		}
+		for _, podName := range kibishiiPodNameList {
+			snapshotContentName, err := util.GetVolumeSnapshotContentNameByPod(client, podName, namespaceBackedUp, backupName)
+			if err != nil {
+				return snapshotCheckPoint, errors.Wrap(err, "Fail get VolumeSnapshotContent name by pod")
+			}
+			snapshotCheckPoint.SnapshotIDList = append(snapshotCheckPoint.SnapshotIDList, strings.Replace(snapshotContentName, "snapcontent", "snapshot", 1))
+			fmt.Println(snapshotCheckPoint)
+		}
+		if len(snapshotCheckPoint.SnapshotIDList) != expectCount {
```
Comment (blackpiglet) [inline test/e2e/util/velero/velero_utils.go:829]: Except the count of VolumeSnapshots, also need to verify VolumeSnapshots' ReadyToUse in Status is set to true

File: test/e2e/util/csi/common.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+	snapshotterClient, err := snapshotterClientSet.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, nil, errors.WithStack(err)
+	}
+
+	return client, snapshotterClient, nil
+}
+
+func GetCsiSnapshotHandleByNamespace(client TestClient, backupName string) ([]string, error) {
```
Comment (blackpiglet) [inline test/e2e/util/csi/common.go:57]: The function' name needs update. VSC is not namespaced.

**Review by blackpiglet — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Pull request merged**